tests/common/example_data_generator/birth_names_generator_factory.py

```diff
@@ -0,0 +1,65 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from tests.common.example_data_generator.birth_names_generator import (
+    BirthNamesGenerator,
+)
+from tests.common.example_data_generator.string_generator_factory import (
+    StringGeneratorFactory,
+)
+
+
+class BirthNamesGeneratorFactory(ABC):
+    __factory: BirthNamesGeneratorFactory
+
+    @abstractmethod
+    def _make(self) -> BirthNamesGenerator:
+        pass
+
+    @classmethod
+    def make(cls) -> BirthNamesGenerator:
+        return cls._get_instance()._make()
+
+    @classmethod
+    def set_instance(cls, factory: BirthNamesGeneratorFactory) -> None:
+        cls.__factory = factory
```

Comment:
this doesn't appear to be used anywhere; is it needed?

Reply:
replied below